src/main/java/org/javarosa/core/model/ItemsetBinding.java

```diff
@@ -56,54 +70,198 @@
+                for (Selection selection : (List<Selection>) rawValue.getValue()) {
+                    currentAnswersInNewChoices.put(selection.choice != null ? selection.choice.getValue() : selection.xmlValue, false);
+                }
+            } else {
+                currentAnswersInNewChoices.put(rawValue.getDisplayText(), false);
+            }
+        }
+
+        return currentAnswersInNewChoices;
+    }
+
+    private void updateQuestionAnswerInModel(FormDef formDef, TreeReference curQRef, Map<String, Boolean> currentAnswersInNewChoices) {
+        IAnswerData rawValue = formDef.getMainInstance().resolveReference(curQRef).getValue();
+
+        if (currentAnswersInNewChoices != null && currentAnswersInNewChoices.containsValue(false)) {
+            IAnswerData filteredAnswer;
+            if (rawValue instanceof MultipleItemsData) {
+                filteredAnswer = getFilteredSelections((MultipleItemsData) rawValue, currentAnswersInNewChoices);
+            } else {
+                filteredAnswer = new StringData("");
+            }
+
+            formDef.getMainInstance().resolveReference(curQRef).setAnswer(filteredAnswer);
+        }
+    }
+
+    /**
+     * @param selections          an answer to a multiple selection question
+     * @param shouldKeepSelection maps each value that could be in @{code selections} to a boolean representing whether
+     *                            or not it should be kept
+     * @return a copy of {@code selections} without the values that were mapped to false in {@code shouldKeepSelection}
+     */
+    private static MultipleItemsData getFilteredSelections(MultipleItemsData selections, Map<String, Boolean> shouldKeepSelection) {
+        List<Selection> newSelections = new ArrayList<>();
+        for (Selection oldSelection : (List<Selection>) selections.getValue()) {
+            String key = oldSelection.choice != null ? oldSelection.choice.getValue() : oldSelection.xmlValue;
+            if (shouldKeepSelection.get(key)) {
+                newSelections.add(oldSelection);
+            }
+        }
+
+        return new MultipleItemsData(newSelections);
+    }
+
+    private Long resolveRandomSeed(DataInstance model, EvaluationContext ec) {
+        if (randomSeedNumericExpr != null)
+            return ((Double) randomSeedNumericExpr.eval(model, ec)).longValue();
+        if (randomSeedPathExpr != null)
+            return toNumeric(randomSeedPathExpr.eval(model, ec)).longValue();
+        return null;
     }
 
     public void localeChanged(String locale, Localizer localizer) {
-        if (choices != null) {
-            for (int i = 0; i < choices.size(); i++) {
-                choices.get(i).localeChanged(locale, localizer);
+        if (latestFilteredChoiceList != null) {
+            for (int i = 0; i < latestFilteredChoiceList.size(); i++) {
+                latestFilteredChoiceList.get(i).localeChanged(locale, localizer);
```

Comment:
New-style `for` might be easier to read.
```
for (SelectChoice choice : latestFilteredChoiceList) {
    choice.localeChanged(locale, localizer);
}
```

Reply:
Yes, can update.